streams/shared/src/main/scala/zio/stream/ZChannel.scala

```diff
@@ -1579,7 +1579,8 @@ object ZChannel {
 
   private[zio] final case class Read[Env, InErr, InElem, InDone, OutErr, OutErr2, OutElem, OutElem2, OutDone, OutDone2](
     more: InElem => ZChannel[Env, InErr, InElem, InDone, OutErr2, OutElem2, OutDone2],
-    done: Fold.K[Env, InErr, InElem, InDone, OutErr, OutErr2, OutElem2, OutDone, OutDone2]
```

Comment:
It was kind of useless to instantiate a `Fold.K` for each `Read`

Reply:
I'm kind of worried about binary compatibility here. Can we expose an auxiliary constructor that takes `done: Fold.K[...]` as an argument as well as a `done` method and mark both  as deprecated? It still won't be fully bincompat due to `unapply`, `copy` etc but it'll reduce the risk significantly

Reply:
Also maybe make all of these case classes package-private to `stream` instead, I don't see a reason why they should be private to `zio`